tmf2/org.lttng.scope.tmf2.views.ui/src/org/lttng/scope/tmf2/views/ui/timeline/widgets/timegraph/TimeGraphWidget.java

```diff
@@ -974,12 +978,33 @@ public void zoom(@Nullable Double pivotX, boolean zoomIn) {
 
             /* Send a corresponding window-range signal to the control */
             TimeGraphModelControl control = getControl();
-            TimeRange range = getViewContext().getCurrentVisibleTimeRange();
-            /* Shrink the time range by half the ZOOM_FACTOR on each side */
-            double newRange = range.getDuration() * (1.0 / newScaleFactor);
-            double diff = range.getDuration() - newRange;
-            long newStart = range.getStart() + Math.round(diff / 2.0);
-            long newEnd = range.getEnd() - Math.round(diff / 2.0);
+            TimeRange visibleRange = getViewContext().getCurrentVisibleTimeRange();
+
+            TimeRange currentSelection = control.getViewContext().getCurrentSelectionTimeRange();
```

Comment:
you can do `getViewContext()` directly, I added a getter at some point since it's so commonly used.